### DISCUSSION TITLE
Error: getaddrinfo EAI_AGAIN redis

### DISCUSSION THREAD

**Discussion by bugacha** [immich-app/immich#13949] [Q&A]
### The bug

I'm trying to install docker image of immich and following steps from https://immich.app/docs/install/docker-compose
I guess latest release right now is v1.119.1

`docker compose up -d` starts up 4 containers fine but immich server isnt working

> # docker ps
> 61eccee52983   ghcr.io/immich-app/immich-machine-learning:release   "./start.sh"             18 minutes ago       Up 18 minutes                                                                      immich_machine_learning
> 67dcd74d873b   redis                                                "redis-server"           About a minute ago   Up About a minute                                                                  immich_redis
> 1ed9e5dacd98   tensorchord/pgvecto-rs                               "postgres -c shared_…"   About a minute ago   Up About a minute                                                                  immich_postgres
> a0dc8bb7a6c4   ghcr.io/immich-app/immich-server:release             "start.sh"               About a minute ago   Up 25 seconds       2283/tcp                                                       immich_server

The issue is that it can't find redis hostname

`Error: getaddrinfo EAI_AGAIN redis
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26) {
  errno: -3001,
  code: 'EAI_AGAIN',
  syscall: 'getaddrinfo',
  hostname: 'redis'
}`

If go into docker's redis, hostname is set to `redis` - so not sure why it doesn't see it

### The OS that Immich Server is running on

Ubuntu 24.10

### Version of Immich Server

v1.119.1

### Version of Immich Mobile App

v1.119.1

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Default downloaded for respective version
wget -O docker-compose.yml https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

REDIS_HOSTNAME=redis

# The location where your uploaded files are stored
UPLOAD_LOCATION=/data/immich/upload
# The location where your database files are stored
DB_DATA_LOCATION=/data/postgres

# To set a timezone, uncomment the next line and change Etc/UTC to a TZ identifier from this list: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
TZ=Europe/Amsterdam

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
# Please use only the characters `A-Za-z0-9`, without special characters or spaces
DB_PASSWORD=postgres

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=postgres
DB_DATABASE_NAME=immich
```

### Reproduction steps

1. Follow steps https://immich.app/docs/install/docker-compose
2. docker compose up -d
3. immich_server isnt working as can't find redis hostname

### Relevant log output

```shell
Initializing Immich v1.119.1
Detected CPU Cores: 16
Starting api worker
Starting microservices worker
Error: getaddrinfo EAI_AGAIN redis
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26) {
  errno: -3001,
  code: 'EAI_AGAIN',
  syscall: 'getaddrinfo',
  hostname: 'redis'
}
```

**Comment by bo0tzz** [comment]
What do the redis logs say?

**Reply by bugacha** [reply]
Redis starts up fine, no errors in logs and I can see it listens n 6379 port.

I think the issue is coming from the fact that I have podman on host OS as well. 
I just tried to install immich docker on a fresh host without podman and it works fine.

My wild guess, podman somehow overwrites docker's network settings and even though immich creates a network for its services and I see interface being created, DNS lookups do not work. Let me dig further.

**Reply by bugacha** [reply]
Unfortunately podman and docker do not work well together on same box. Close it please.
